### PR TITLE
Support shared named environments during startup

### PR DESCRIPTION
**Pull request by oschulz**
Implements #35354 .

I hope I did this right - from what I saw, most things @fredrikekre had in his proposed patch in #35354, `load_path_expand()` does already now. So just using `load_path_expand()` instead of `expanduser()` in `init_active_project()` seems to do the trick. I'm not an expert on Julia's init process, though.

WIth this, both `--project=@myenv` and `JULIA_PROJECT=@myenv` seem to behave the same way as `activate @myenv` on the Pkg console.

### CODE REVIEW

**Comment by oschulz** [comment]
Ah, not quite that simple it seems (see failed tests) ...

**Comment by oschulz** [comment]
Got tests to pass - @vtjnash , are the additional tests sufficient?

File: base/initdefs.jl
Comment (vtjnash) [inline base/initdefs.jl:239]:
```suggestion
        startswith(project, "@") ? load_path_expand(project) : abspath(expanduser(project))
```

Looking at `load_path_expand`, it already calls `current_project` and `abspath`, and might return `nothing`

**Review by vtjnash — APPROVED** [review]

**Review by fredrikekre — APPROVED** [review]
This doesn't work on Windows? I see no reason it shouldn't, and in that case the new tests should be moved out from the `if`.

**Comment by oschulz** [comment]
Good point - why are the previous test cases in `if !Sys.iswindows()`, though?

**Comment by fredrikekre** [comment]
Because they test expansion of `~`.

**Comment by oschulz** [comment]
Ahh, silly me. Ok, I'll move it out.

**Comment by oschulz** [comment]
So it fails on Windows with

```
  Expression: occursin(expanded, readchomp(setenv(`$exename -E 'Base.active_project()'`, "JULIA_PROJECT" => "@foo", "HOME" => homedir())))
   Evaluated: occursin("C:\\Windows\\system32\\config\\systemprofile\\.julia\\environments\\foo\\Project.toml", "\"C:\\\\Windows\\\\system32\\\\config\\\\systemprofile\\\\.julia\\\\environments\\\\foo\\\\Project.toml\"")
```

But where's that extra level of `\\` escaping coming from?

**Comment by oschulz** [comment]
Ok, tests are much nicer now.

**Comment by oschulz** [comment]
Question: `load_path_expand(project)` returns the full project path including `"/Project.toml"` and `expanduser(project)` just returns the directory. Is that Ok (same mixed behavior as before) or should we harmonize this so that `init_active_project()` always returns the project with or without `"/Project.toml"` at the end?

**Comment by oschulz** [comment]
@vtjnash , is this good to merge as is, for you?

**Comment by fredrikekre** [comment]
LGTM, but might be news-worthy?

**Comment by oschulz** [comment]
Good point - under "Command-line option changes", I guess?

**Comment by oschulz** [comment]
I added a news item - text Ok like that?

**Comment by musm** [comment]
@oschulz  Was the force-push just a documentation change? I want to confirm I can merge this if any code-changes were made after the approvals.

**Comment by oschulz** [comment]
> @oschulz Was the force-push just a documentation change?

**Comment by oschulz** [comment]
> @oschulz Was the force-push just a documentation change?

Oh, sorry - just wanted to present a cleaner history. No, the force push included improvements to the tests that I added after @fredrikekre initial approval (since he suggested to also test on Windows).

**Comment by oschulz** [comment]
Thanks!

**Comment by oschulz** [comment]
I should have done this a few weeks earlier, in time for v1.6. :-)

**Comment by musm** [comment]
That gives you a reason to look forward to 1.7 😉

**Comment by oschulz** [comment]
;-)

**Pull request merged**
